src/cmd/flux-jobs.py

```diff
@@ -198,6 +198,13 @@ def __call__(self, parser, namespace, values, option_string=None):
         setattr(namespace, "filtered", True)
 
 
+def JobID_check(arg):
```

Comment:
It would be better to do this in the `JobID` constructor instead. If we raise `ValueError` directly, then `argparse` will kind of do the right thing automatically wherever the type of an argument is `JobID`, e.g.

```diff
diff --git a/src/bindings/python/flux/job/JobID.py b/src/bindings/python/flux/job/JobID.py
index 2debc6f3e..a0d8dd3e5 100644
--- a/src/bindings/python/flux/job/JobID.py
+++ b/src/bindings/python/flux/job/JobID.py
@@ -55,7 +55,10 @@ class JobID(int):
         if isinstance(value, int):
             jobid = value
         else:
-            jobid = id_parse(value)
+            try:
+                jobid = id_parse(value)
+            except OSError:
+                raise ValueError(f"{value} is not a valid Flux jobid")
         self = super(cls, cls).__new__(cls, jobid)
         self.orig_str = str(value)
         return self
```

Then:

```console
$ flux jobs g1234
usage: flux-jobs [-h] [-a] [-A] [-c N] [-f STATE|RESULT] [-n]
                 [-u [USERNAME|UID]] [-o FORMAT] [--color WHEN] [-R] [-L N]
                 [--recurse-all] [--threads N] [--stats] [--stats-only]
                 [JOBID [JOBID ...]]
flux-jobs: error: argument JOBID: invalid JobID value: 'g1234'
```